[Joan] IP

sjoann opened the pull request:
### Duke LITE

> “Productivity is being able to do things that you were never able to do before.” – Franz Kafka [source](https://www.chanty.com/blog/productivity-quotes/#:~:text=%E2%80%9CProductivity%20is%20being%20able%20to,in%20not%20reaching%20your%20goal.)

Duke LITE frees your mind of having to remember things you need to do. It's,

- Text-based
- Easy to pick up
- SUPER FAST TO USE

All you need to do is,

1. Add your tasks.
2. let it manage your tasks for you 😉

And it is **FREE!**

Features:

- [ ] Add or Delete tasks: Todo, Deadlines and Events
- [ ] Managing deadlines dates
- [x] Mark or unmark your tasks
- [ ] Enumerate your list of tasks
- [ ] Find your task

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

`public static void main(String[] args) throws IOException {
        new Duke("data/tasks.txt").run();
}`

Commit:
Add Gradle support

sjoann committed:
Add Level-1

sjoann committed:
Add Level-2

sjoann committed:
Add Level-3

sjoann committed:
Add Level-4

sjoann committed:
Add A-TextUiTesting

sjoann committed:
Add Level-5

sjoann committed:
Add Level-6

sjoann committed:
Add Level 7

sjoann committed:
Add Level-7

sjoann committed:
Add Level-8

sjoann committed:
Update Level-7

sjoann committed:
Update Level-8

sjoann committed:
Add A-MoreOOP

sjoann committed:
Add A-Packages

sjoann committed:
Merge pull request #1 from nus-cs2103-AY2223S1/add-gradle-support

Add Gradle support

sjoann committed:
Add A-JUnit

sjoann committed:
Add A-Jar

sjoann committed:
Add AJavaDoc

sjoann committed:
Add A-CodingStandard

sjoann committed:
Add Level-9

sjoann committed:
Fix merge conflicts

sjoann committed:
Merge branch 'branch-Level-9'

sjoann committed:
Fix merge conflict

sjoann committed:
Fix conflict

sjoann committed:
Fix bugs from merge conflict

sjoann committed:
fix conflicts and bugs

sjoann committed:
Fix merge conflicts

sjoann committed:
Add gradlew checkstyle

sjoann committed:
Update tasks

sjoann committed:
Edit code style

sjoann committed:
Add Level-10

sjoann committed:
Remove unneccesary code

sjoann committed:
Update code style

sjoann committed:
Fix coding style violation

sjoann committed:
Fix delete command

jicsontoh commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,61 @@
+        db.flip();
+        return db;
+    }
+}
```
Maybe should leave a new line at the end of file?

jicsontoh commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,37 @@
+     * @param taskList TaskList object containing ArrayList of Task.
+     * @param ui Ui object.
+     * @param storage Storage object.
+     * @return String.
```
Perhaps can be more descriptive in the return type?

jicsontoh commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,102 @@
+    /**
+     * Returns a LocalDateTime object.
+     *
+     * @param date String of the date with format dd/mm/yyyy 0000.
```
Good description of variables with clear format!

jicsontoh commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,54 @@
+import java.io.IOException;
+
+
+/** Contains storage, tasks and ui to help interact with user and execute commands */
```
Clear and concise description of class and methods!

jicsontoh reviewed (commented):
Overall good job in programming Duke, not much issues except those mentioned in the comments! :)

sjoann committed:
Fix coding style

sjoann committed:
Add A-Assertions

sjoann committed:
Merge pull request #2 from sjoann/branch-A-Assertions

Add A-Assertions

sjoann committed:
Improve code quality

sjoann committed:
Merge pull request #3 from sjoann/branch-A-CodeQuality

Improve code quality

sjoann committed:
Find Command: make date searching available.

Find Command allows searching of tasks based on task
description.

Allowing users to search tasks on a specific date will
make the GUI more user-friendly.

sjoann committed:
Count Command: count number of completed or uncompleted tasks.

It provides a way to leverage statistics about the items
managed by the App.

Users can input "count completed" and "count uncompleted"
to know the number of completed and uncompleted tasks
respectively.

sjoann committed:
Add A-CI

sjoann committed:
Add A-UserGuide

sjoann committed:
Change mainClassName

sjoann committed:
Change mainClassName

sjoann committed:
Update README

sjoann committed:
Update GUI

sjoann committed:
Update error message

sjoann committed:
Remove dead code

sjoann committed:
Switch import statements order

sjoann committed:
Change naming convention

sjoann committed:
Update bye

sjoann committed:
Add more testing
